pose_lookup/plugin.cpp

```diff
@@ -4,139 +4,189 @@
 #include "common/pose_prediction.hpp"
 #include "common/data_format.hpp"
 #include "common/plugin.hpp"
+#include "common/global_module_defs.hpp"
 
-/*pyh: reusing data_loading from ground_truth_slam*/
+
+#include "utils.hpp"
 #include "data_loading.hpp"
 
 using namespace ILLIXR;
 
+
 class pose_lookup_impl : public pose_prediction {
 public:
     pose_lookup_impl(const phonebook* const pb)
-		: sb{pb->lookup_impl<switchboard>()}
-		, _m_sensor_data{load_data()}
-		, dataset_first_time{_m_sensor_data.cbegin()->first}
-		, _m_start_of_time{std::chrono::high_resolution_clock::now()}
-		, _m_vsync_estimate{sb->subscribe_latest<time_type>("vsync_estimate")}
+        : sb{pb->lookup_impl<switchboard>()}
+        , _m_sensor_data{load_data()}
+        , dataset_first_time{_m_sensor_data.cbegin()->first}
+        , _m_start_of_time{std::chrono::high_resolution_clock::now()}
+        , _m_vsync_estimate{sb->subscribe_latest<time_type>("vsync_estimate")}
+        /// TODO: Set with #198
+        , enable_alignment{ILLIXR::str_to_bool(getenv_or("ILLIXR_ALIGNMENT_ENABLE", "False"))}
+        , init_pos_offset{0}
+        , align_rot{Eigen::Matrix3f::Zero()}
+        , align_trans{0}
+        , align_quat{0}
+        , align_scale{0.0}
+        , path_to_alignment{ILLIXR::getenv_or("ILLIXR_ALIGNMENT_FILE", "./metrics/alignMatrix.txt")}
```

Comment:
Who sets this?

Reply:
This is set right here, given that no environment variable will exist until set via #198. A transitional hook.

Reply:
Yeah, I saw the #198 comment but I meant was this (sorry, should have clarified): the enabling/disabling flag is already set via the config and runner. Why not do the same for the path (here and in offload_data)? At a minimum, it would at least save recompilation.

Reply:
These variables can still be set easily by adding an `export ILLIXR_ALIGNMENT_FILE=...` in the user's environment via the shell or in `runner.sh` before the `python` command.

Any flags set pre-#198 by the config need to be added to the schema and the runner environment variable overrides. These changes would then have to be reverted once #198 is merged. I let it be for the flags, since those had already been implemented.

Reply:
Okay, that makes sense. I guess that just leaves the variable setting for `load_monado` and the print guard for the fix PR. @Boyuan-Tian